packages/cmf/src/middlewares/http/middleware.js

```diff
@@ -122,40 +122,47 @@ export const httpMiddleware = ({ dispatch }) => next => (action) => {
 	if (!isHTTPRequest(action)) {
 		return next(action);
 	}
-	const config = mergeOptions(action);
-	dispatch(httpRequest(action.url, config));
-	if (action.onSend) {
+	const httpAction = get(action, 'cmf.http', action);
+	const config = mergeOptions(httpAction);
+	dispatch(httpRequest(httpAction.url, config));
+	if (httpAction.onSend) {
 		dispatch({
-			type: action.onSend,
-			action,
+			type: httpAction.onSend,
+			httpAction,
 		});
 	}
 	const onHTTPError = (error) => {
-		const newAction = Object.assign({
-			error: {
-				name: error.name,
-				message: error.description || error.message,
-				number: error.number,
-				stack: error.stack,
-			},
-		}, action);
-
-		// clone the response object else the next call to text or json
-		// triggers an exception Already use
-		newAction.error.stack.response.clone().text().then((response) => {
-			try {
-				newAction.error.stack.response = response;
-				newAction.error.stack.messageObject = JSON.parse(response);
-			} finally {
-				if (newAction.onError) {
-					dispatch(onError(newAction, newAction.error));
-				} else {
-					dispatch(httpError(newAction.error));
+		const errorObject = {
+			name: error.name,
+			message: error.description || error.message,
+			number: error.number,
+			stack: error.stack,
+		};
+		const clone = get(error, 'stack.response.clone');
+		if (!clone) {
+			dispatch({
+				type: 'HTTP_REDUCE_ERROR',
```

Comment:
at the moment when we have a JS error in a middleware it's hidden by can't read clone attribute of undefined.
So let's dispatch that error so we may display the error 'to the end user or not'

May I just throw the error ?